### ISSUE TITLE
Build issues with interface newer than 0.1.5

### ISSUE THREAD

**Issue by westekim** [liodali/osm_flutter#186]
I ran into build troubles. With 0.22.3 of the plugin and anything newer than 0.1.5 of the interface, it won't build. But with 0.22.3 and 0.1.5 it builds and seems to work. 

With 0.22.3 I get this error message on build:

```
../../AppData/Local/Pub/Cache/hosted/pub.dartlang.org/flutter_osm_plugin-0.22.3/lib/src/controller/osm/osm_controller.dart:10:7: Error: The non-abstract class 'MobileOSMController' is missing implementations for these members:
 - IBaseOSMController.getMapCenter
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class MobileOSMController extends IBaseOSMController {
      ^^^^^^^^^^^^^^^^^^^
../../AppData/Local/Pub/Cache/hosted/pub.dartlang.org/flutter_osm_interface-0.1.9+1/lib/src/osm_controller/osm_controller.dart:82:20: Context: 'IBaseOSMController.getMapCenter' is defined here.
  Future<GeoPoint> getMapCenter();
                   ^^^^^^^^^^^^
../../AppData/Local/Pub/Cache/hosted/pub.dartlang.org/flutter_osm_plugin-0.22.3/lib/src/controller/osm/osm_controller.dart:303:16: Error: The method 'MobileOSMController.setIconStaticPositions' has fewer named arguments than those of overridden method 'IBaseOSMController.setIconStaticPositions'.
  Future<void> setIconStaticPositions(
               ^
../../AppData/Local/Pub/Cache/hosted/pub.dartlang.org/flutter_osm_interface-0.1.9+1/lib/src/osm_controller/osm_controller.dart:41:16: Context: This is the overridden method ('setIconStaticPositions').
  Future<void> setIconStaticPositions(
               ^
```

Is the problem in my environment or in the plugin?

**Comment by liodali** [comment]
no,
but i don't why in pubspeci i already specified that should depend on 0.1.5 but when pub get download dependencies downlad last version 0.1.9+1
you can override dependencies this problem will be fix in 0.25.0
but for now add this in pubspec 
```yaml
dependency_overrides:
  flutter_osm_interface: "0.1.5"
```
and run in cmd line
`flutter clean && flutter pub get`

**Comment by westekim** [comment]
Thanks for the quick reply! 
This workaround works (as already mentioned in the first post)
Good to know the error was not on my end and that 0.22.3 should depend on the interface version I found to work.
